internal/pkg/consensus/block_validation_test.go

```diff
@@ -29,33 +33,163 @@ func TestBlockValidSemantic(t *testing.T) {
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "non-account actor")
+	})
+
+	t.Run("accepts block with bls messages in monotonic sequence", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			blsMessages: []*types.UnsignedMessage{msg1, msg2, msg3},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.NoError(t, err)
+	})
+
+	t.Run("accepts block with secp messages in monotonic sequence", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			secpMessages: []*types.SignedMessage{{Message: *msg1}, {Message: *msg2}, {Message: *msg3}},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.NoError(t, err)
+	})
+
+	t.Run("rejects block with messages out of order", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			blsMessages: []*types.UnsignedMessage{msg1, msg3, msg2},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "out of order")
+	})
+
+	t.Run("rejects block with gaps", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			blsMessages: []*types.UnsignedMessage{msg1, msg3},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "out of order")
+	})
+
+	t.Run("rejects block with bls message with nonce too low", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			blsMessages: []*types.UnsignedMessage{msg0},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "out of order")
+	})
+
+	t.Run("rejects block with secp message with nonce too low", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			secpMessages: []*types.SignedMessage{{Message: *msg0}},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "out of order")
+	})
+
+	t.Run("rejects block with message too high", func(t *testing.T) {
+		validator := consensus.NewDefaultBlockValidator(mclock, &fakeMsgSource{
+			blsMessages: []*types.UnsignedMessage{msg2},
+		}, &fakeChainState{
+			actor: newActor(t, 0, 2),
+		})
+
+		err := validator.ValidateMessagesSemantic(ctx, c, parents.Key())
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "out of order")
+	})
+}
```

Comment:
Lovely cases. Could you add one with SECP sequence < BLS sequence for same actor (fails validation) and the reverse (passes)?